### ISSUE TITLE
Double-click on the sample waveform to open the sample editor

### ISSUE THREAD

**Issue by trebmuh** [hydrogen-music/hydrogen#455]
I'd suggest that a double-click on the sample waveform (see picture below) to open the "edit layer" action. Even if it's not here (yet), I'm 90% of the time doing it (certainly a Pavlov thing) so I'm guessing it makes sense from a user-friendly point of view.

![sampleeditor-h2](https://cloud.githubusercontent.com/assets/8705846/21137174/f84e1cde-c128-11e6-84c2-14c25e9a3399.png)

What do you guys think about this?

**Comment by alworx** [comment]
I second that :)
Maybe you and/or mauser want to have a look and test my first contribution to this wonderful project: https://github.com/alworx/hydrogen/commit/e898cbc81f2486c8add5ed0cc9620a8c4331678b

(Sorry for the commit-mess, had some troubles with github/user management)

**Comment by xjjx** [comment]
+1 from me for the idea

Try that .. IMHO much more readable ;-)
BTW. are you sure there shouldn't be:
if( ! pCompo ) return;

?

```C
void InstrumentEditor::waveDisplayDoubleClicked( QWidget* pRef )
{		
	if ( ! m_pInstrument ) return;

	InstrumentComponent* pCompo = m_pInstrument->get_component(m_nSelectedComponent);
	if( pCompo ) return;
	
	H2Core::InstrumentLayer *pLayer = pCompo->get_layer( m_nSelectedLayer );
	if ( pLayer ) {
		Sample* pSample = pLayer->get_sample();
		if( ! pSample ) return;

		QString name = pSample->get_filepath();
		HydrogenApp::get_instance()->showSampleEditor( name, m_nSelectedComponent, m_nSelectedLayer );
	} else {
		loadLayer();
	}	
}
```

**Comment by pvint** [comment]
I don't have time to test, but I had a quick look and I have one comment, @alworx :

In:
`connect( m_pWaveDisplay, SIGNAL( doubleClicked(QWidget*) ), this, SLOT( waveDisplayDoubleClicked(QWidget*) ) );`

I seem to recall there being an issue with using the doubleClicked signal (not sure though!). Maybe it's not supported in different versions of QT or something?  I think I tried this before but it did not work for me, but maybe I just did it wrong. ;)

Overall, looks about right, but I just wanted to raise that question (and perhaps someone else has some input). I will test when I can.

Thanks for the work!
Paul

**Comment by alworx** [comment]
> Try that .. IMHO much more readable ;-)
A choice of flavour, I'd say - I'd agree that your proposal is more readable, but I usually value consistent coding style (per file, at least) higher than flavour - especially this code-block is stolen from buttonClicked() some lines below my code ;)

> if( ! pCompo ) return;
Didn't find that line in code, just in your proposal - can you point me to the source?

> I seem to recall there being an issue with using the doubleClicked signal (not sure though!).
Doubleclicks can cause troubles when single clicks are also handled in the same component, but as I'm pretty unexperienced with Qt I'd be happy if someone more involved points me in the right direction!

**Comment by xjjx** [comment]
right - my fault. It should be:
```C
void InstrumentEditor::waveDisplayDoubleClicked( QWidget* pRef )
{		
	if ( ! m_pInstrument ) return;

	InstrumentComponent* pCompo = m_pInstrument->get_component(m_nSelectedComponent);
	if( ! pCompo ) return;
	
	H2Core::InstrumentLayer *pLayer = pCompo->get_layer( m_nSelectedLayer );
	if ( pLayer ) {
		Sample* pSample = pLayer->get_sample();
		if( ! pSample ) return;

		QString name = pSample->get_filepath();
		HydrogenApp::get_instance()->showSampleEditor( name, m_nSelectedComponent, m_nSelectedLayer );
	} else {
		loadLayer();
	}	
}
```

**Comment by alworx** [comment]
@xjjx: Just following the [coding conventions](https://github.com/hydrogen-music/hydrogen/blob/master/DEVELOPERS), but I'm perfectly fine with your codesnippet - I'll issue a pull request and let the maintainer decide :)

**Comment by trebmuh** [comment]
Nice to see that you are working on it, thanks for that.
I'll comment on #491 .

**Comment by trebmuh** [comment]
Implemented in #491.
Closing.
